design-toolkit/components/src/components/tree/expand-toggle.tsx

```diff
@@ -0,0 +1,70 @@
+// __private-exports
+/*
+ * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { TreeStyles, TreeStylesDefaults } from '@/components/tree/styles';
+import { isSlottedContextValue } from '@/lib/utils';
+import ChevronDown from '@accelint/icons/chevron-down';
+import ChevronUp from '@accelint/icons/chevron-up';
+import { type ForwardedRef, forwardRef, useContext, useMemo } from 'react';
+import { ButtonContext, useContextProps } from 'react-aria-components';
+import { Button } from '../button';
+import type { ButtonProps } from '../button/types';
+import { Icon } from '../icon';
+import { TreeContext } from './';
+
+const { spacing, expansion } = TreeStyles();
+
+type ExpandToggleProps = ButtonProps & {
+  isExpanded: boolean;
+  isVisible: boolean;
+  isViewable: boolean;
+  hasChildItems: boolean;
+};
+
+export const ExpandToggle = forwardRef(
```

Comment:
With React 19, we don't want to use forwardRef anymore. Ref is now just a regular prop, you can use RefAttributes type to add it if dealing with underlying RAC prop types, otherwise use ComponentPropsWithRef